### DISCUSSION TITLE
Fonts with names containing " Cond" cannot be used

### DISCUSSION THREAD

**Discussion by hyunrealshadow** [typst/typst#2921] [Q&A]
Sorry I can't write the full name of the font for privacy reasons, it's something like `XXX Sans`

I put the font containing only the standard font name (XXX Sans) into the fonts folder, and the output is as follows
```
./typst fonts --font-path ~/fonts --variants
XXX Sans
- Style: Italic, Weight: 400, Stretch: FontStretch(1000)
- Style: Normal, Weight: 700, Stretch: FontStretch(1000)
- Style: Italic, Weight: 300, Stretch: FontStretch(1000)
- Style: Italic, Weight: 900, Stretch: FontStretch(1000)
- Style: Normal, Weight: 400, Stretch: FontStretch(1000)
- Style: Normal, Weight: 300, Stretch: FontStretch(1000)
- Style: Italic, Weight: 700, Stretch: FontStretch(1000)
- Style: Normal, Weight: 900, Stretch: FontStretch(1000)
- Style: Normal, Weight: 500, Stretch: FontStretch(1000)
- Style: Italic, Weight: 500, Stretch: FontStretch(1000)
```
Put the fonts that only contain standard font names (XXX Sans) and condensed name (XXX Sans Cond) into the fonts folder, and the output is as follows
```
./typst fonts --font-path ~/fonts --variants
XXX Sans
- Style: Italic, Weight: 400, Stretch: FontStretch(1000)
- Style: Normal, Weight: 700, Stretch: FontStretch(1000)
- Style: Normal, Weight: 400, Stretch: FontStretch(1000)
- Style: Italic, Weight: 400, Stretch: FontStretch(1000)
- Style: Italic, Weight: 300, Stretch: FontStretch(1000)
- Style: Italic, Weight: 900, Stretch: FontStretch(1000)
- Style: Normal, Weight: 400, Stretch: FontStretch(1000)
- Style: Normal, Weight: 300, Stretch: FontStretch(1000)
- Style: Italic, Weight: 700, Stretch: FontStretch(1000)
- Style: Normal, Weight: 700, Stretch: FontStretch(1000)
- Style: Normal, Weight: 900, Stretch: FontStretch(1000)
- Style: Italic, Weight: 700, Stretch: FontStretch(1000)
- Style: Normal, Weight: 500, Stretch: FontStretch(1000)
- Style: Italic, Weight: 500, Stretch: FontStretch(1000)
```
Put the font containing only the condensed name (XXX Sans Cond) into the fonts folder, the output is as follows
```
./typst fonts --font-path ~/fonts --variants
XXX Sans
- Style: Normal, Weight: 400, Stretch: FontStretch(1000)
- Style: Italic, Weight: 400, Stretch: FontStretch(1000)
- Style: Normal, Weight: 700, Stretch: FontStretch(1000)
- Style: Italic, Weight: 700, Stretch: FontStretch(1000)
```
In Windows systems, the font looks like this
![image](https://github.com/typst/typst/assets/36305150/53887290-dfd3-4cac-8d4b-066f509a34c6)
![image](https://github.com/typst/typst/assets/36305150/1d9532ee-9084-49ce-b921-3bd6e7e1b60f)
Some fonts in the red box cannot be used

When I used Font Forge to change the font name to XXX SansCond, it worked(I didn't change the italicised fonts)
Now it'll show up as XXX SansCond
```
XXX Sans
- Style: Italic, Weight: 400, Stretch: FontStretch(1000)
- Style: Normal, Weight: 700, Stretch: FontStretch(1000)
- Style: Italic, Weight: 300, Stretch: FontStretch(1000)
- Style: Italic, Weight: 900, Stretch: FontStretch(1000)
- Style: Normal, Weight: 400, Stretch: FontStretch(1000)
- Style: Normal, Weight: 300, Stretch: FontStretch(1000)
- Style: Italic, Weight: 700, Stretch: FontStretch(1000)
- Style: Normal, Weight: 900, Stretch: FontStretch(1000)
- Style: Normal, Weight: 500, Stretch: FontStretch(1000)
- Style: Italic, Weight: 500, Stretch: FontStretch(1000)
XXX SansCond
- Style: Normal, Weight: 700, Stretch: FontStretch(1000)
- Style: Normal, Weight: 400, Stretch: FontStretch(1000)
```
In Windows it now has the following name
![image](https://github.com/typst/typst/assets/36305150/10c3f6a8-c171-45a2-b6e9-ce36c02634f1)
Two fonts can be seen in Word
![image](https://github.com/typst/typst/assets/36305150/b7ec7984-49cc-4e39-bed8-a1052c61d63d)

The `stretch` parameter does not work in this case
I'm not sure if this is an issue, so use discussion to ask the question

**Comment by laurmaedje** [comment]
The current approach is a best effort way of dealing with the mess that is font naming and metadata. It is certainly not ideal. Part of the solution is likely this: #2098.

**Reply by hyunrealshadow** [reply]
Thanks for the reply, looking forward to #2098 early release, as an temporary solution I'll use FontForge to change the font name to fix the problem
